### DISCUSSION TITLE
Unknown at rule @tailwindcss(unknownAtRules)

### DISCUSSION THREAD

**Discussion by huntedking150** [tailwindlabs/tailwindcss#13881] [Help] [answered]
I am following all the steps outlined in the installation guide on the Tailwind CSS website. However, I consistently encounter the error shown in the screenshot. While some of the code functions correctly and doesn't completely crash, Tailwind CSS does not produce the expected effects.
![Screenshot from 2024-06-22 16-24-17](https://github.com/tailwindlabs/tailwindcss/assets/140186099/f3dbb5e8-bff3-4807-bb0b-b1468a9e2534)

I havenot added anything extra code to any file. Just I am just trying to configure tailwind css in my project with default code generated by "npm create vite@latest".

I have used the  following commands:
npm create vite@latest
then I entered project name and selected vite for my project 
cd <projectName>
npm i
npm install -D tailwindcss postcss autoprefixer
npx tailwindcss init -p
then changed the content of tailwind.config.js file
with :
content: [
    "./index.html",
    "./src/**/*.{js,ts,jsx,tsx}",
  ],
added :
@tailwind base;
@tailwind components;
@tailwind utilities;
in index.css 
and I started getting error in my index.css file

**Comment by wongjn** [comment, marked as answer]
It's a warning from VSCode, not explicitly an error. Tailwind should still work regardless. If you really want to get rid of the error though, you could do one of:
- Use the [PostCSS plugin](https://marketplace.visualstudio.com/items?itemName=csstools.postcss)
- Use the [Tailwind intelliSense plugin](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss)
- [Turn off CSS validation](https://code.visualstudio.com/docs/languages/css#_syntax-verification-linting)

**Reply by waxben** [reply]
helped

**Reply by noah-tah** [reply]
worked for me when i installed [Tailwind intelliSense plugin](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss)

**Reply by RN0826** [reply]
thanks bro, it helped me out too,
I was following the installation procedure for magicui and got the same error and it got resolved after installing PortCSS plugin
got the error in globals.css file

**Reply by kellyannclark** [reply]
PostCSS Language Support solved it for me, thank you!

**Reply by shakalaka696** [reply]
Thanks for helping buddy

**Reply by HARSHITHA194** [reply]
Installing PostCSS Language Support worked for me Thank u !

**Reply by mhdshrz** [reply]
For all people that have installed [Tailwind CSS Intellisens VSCode extension](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss) but still get the same warnings (like I did), there's no need to install [PostCSS Language Support VSCode extenstion](https://marketplace.visualstudio.com/items?itemName=csstools.postcss) or [disable CSS syntax verification linting](https://code.visualstudio.com/docs/languages/css#_syntax-verification-linting).
You can add this configuration code to your settings.json (I'd prefer to add it to a local-to-workspace settings.json rather than to the User settings.json, you can do that pressing `Ctrl + Shift + P` to open up the Command Palette and search for `Open Workspace Settings (JSON)`, find it, then select it. It will automatically create a `.vscode` folder inside your project - if there weren't any - and adds your workspace settings.json inside it):
```json
    "files.associations": {
        "*.css": "tailwindcss"
    },
```
Tailwind CSS Intellisense VSCode extension details:
> Use the `files.associations` setting to tell VS Code to always open `.css` files in Tailwind CSS mode.

**Reply by Austin1serb** [reply]
PostCSS plugin turns off linting for my css files. I had to do an extension bisect to figure out what was messing with it

**Reply by mozes22** [reply]
> PostCSS plugin turns off linting for my css files

Same here, what did you do to resolve this?

**Reply by Austin1serb** [reply]
> > PostCSS plugin turns off linting for my css files
> 
> Same here, what did you do to resolve this?

I uninstalled the plugin

**Reply by mozes22** [reply]
> > > PostCSS plugin turns off linting for my css files
> > 
> > 
> > Same here, what did you do to resolve this?
> 
> I uninstalled the plugin

I don't have PostCSS plugin/extension, I only have Tailwind Extension "bradlc.vscode-tailwindcss" and Stylelint extension "stylelint.vscode-stylelint" and as soon as I add the:
```
"files.associations": {
   "css": "tailwindcss"
},
```
in settings.json, the Stylelint linting for all CSS files in my project is turned off.

**Reply by KingPsychopath** [reply]
> ```json
>     "files.associations": {
>         "*.css": "tailwindcss"
>     },
> ```
> 
> Tailwind CSS Intellisense VSCode extension details:
> 
> > Use the `files.associations` setting to tell VS Code to always open `.css` files in Tailwind CSS mode.

Thank you so much.

**Reply by Chatan2700** [reply]
Dud I love you, thank you so much

**Comment by Pyr33x** [comment]
You can easily fix it by setting up a new language mode, `ctrl + shift + p` then type language Mode: `tailwindcss`
and make sure you are setting language mode when you're currently in index.css file

**Reply by muhammadmahfuj** [reply]
thanks a lot. this fixed the issue for me.

**Reply by ADTC** [reply]
You need to install [Tailwind CSS Intellisense extension](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss) first.

**Reply by Anuragh33** [reply]
it works thank you.

**Reply by BlackSheep47** [reply]
thanks it's helped

**Reply by james-do-things** [reply]
that was the trick - thanks!

**Reply by sudokunym** [reply]
thanks! this fixed it for me

**Reply by fallf** [reply]
thank you this worked for

**Reply by AlyssonCaetano** [reply]
that was the trick, this worked for - thanks!

**Reply by Austin1serb** [reply]
great awesome fix man

**Reply by millan-figueroa** [reply]
your suggestion fixed it faster that an hour on chatgpt 

edited to add: just tried this again on a new project and tailwindcss didn't come up on the language list, I figured out that the last time I had installed tailwindcss intellisense extension trying to get rid of that compiler warning. So if it doesn't come up for you when you start typing, try that.

**Reply by Pyr33x** [reply]
> your suggestion fixed it faster that an hour on chatgpt

hahah your welcome

**Reply by Masked-Kunsiquat** [reply]
thank you @Pyr33x !

**Reply by stefania-d-dev** [reply]
Thank you @Pyr33x 🥹

**Reply by Pyr33x** [reply]
> Thank you @Pyr33x 🥹

❤️🙂

**Reply by Codeformage** [reply]
thank you @Pyr33x

**Reply by folsze** [reply]
this change makes css variables be highlighted red for me:

$attribution-height: 30px;

ERROR:
{ expectedtailwindcss(css-curlyexpected)

**Reply by katerlouis** [reply]
@folsze You've probably figured it out yourself already, but in case somebody else is wondering: `$foobar` is not a css variable, but a scss variable.

**Reply by eltonSalanic** [reply]
Amazing, this worked!

**Comment by huntedking150** [comment]
Installing Plugin helped me. Thanks Everyone

**Comment by zaighamhussain786** [comment]
Install plugin **Post CSS Language Support** in VS Code then it'll end

**Reply by SeekyLeaky** [reply]
Thank you, this fixed it for me instantly

**Reply by codebdy** [reply]
thank you

**Comment by Pyr33x** [comment]
permanent fix: just add this to your vscode settings.json

```json
"files.associations": {
  "*.css": "tailwindcss"
},
```

and if it was helpful, please hit the anwser mark

**Reply by jerry-xon** [reply]
Thanks alot this helped me !!!!!

**Reply by NickDatLe** [reply]
> permanent fix: just add this to your vscode settings.json
> 
> ```json
> "files.associations": {
>   "*.css": "tailwindcss"
> },
> ```
> 
> and if it was helpful, please hit the anwser mark

This worked on Cursor AI, thanks!

**Reply by advaitju** [reply]
Just an FYI, you need the [Tailwind CSS Intellisense](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss) VSCode plugin for this to work.

**Reply by mohdfurkan01** [reply]
> permanent fix: just add this to your vscode settings.json
> 
> ```json
> "files.associations": {
>   "*.css": "tailwindcss"
> },
> ```
> 
> and if it was helpful, please hit the anwser mark

Huge thanks! You've definitely lowered my frustration level

**Reply by Pyr33x** [reply]
> > permanent fix: just add this to your vscode settings.json
> > ```json
> > "files.associations": {
> >   "*.css": "tailwindcss"
> > },
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and if it was helpful, please hit the anwser mark
> 
> Huge thanks! You've definitely lowered my frustration level

haha, you're welcome dude. 🫶

**Reply by viszec** [reply]
haha, 100% thanks, this worked on Cursor!

**Reply by jackofsometrade** [reply]
> ```json
> "files.associations": {
>   "*.css": "tailwindcss"
> },
> ```

Nope, this completely disables CSS. All CSS files now look like plain text.

**Reply by aervxa** [reply]
> > ```json
> > "files.associations": {
> >   "*.css": "tailwindcss"
> > },
> > ```
> 
> Nope, this completely disables CSS. All CSS files now look like plain text.

Maybe try installing the intellisense vscode plugin

**Reply by rmitaccs40** [reply]
> permanent fix: just add this to your vscode settings.json
> 
> ```json
> "files.associations": {
>   "*.css": "tailwindcss"
> },
> ```
> 
> and if it was helpful, please hit the anwser mark

Hi, I'm just wonder what is the purpose of this? And in case, if I want to use other CSS framework like Bootstrap. Do I need to remove this?

**Reply by guilleliss** [reply]
Very simple, fixed it instantly, no need to install extensions or anything

**Reply by aksagarwal42** [reply]
> > permanent fix: just add this to your vscode settings.json
> > ```json
> > "files.associations": {
> >   "*.css": "tailwindcss"
> > },
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and if it was helpful, please hit the anwser mark
> 
> Hi, I'm just wonder what is the purpose of this? And in case, if I want to use other CSS framework like Bootstrap. Do I need to remove this?

You can add this to your respective project's or wokrspace's settings json and that will keep the association limited to that particular project or workspace.

**Comment by zaighamhussain786** [comment]
1. **Open VSCode**.

2. **Access Settings**:
   - Click on the gear icon (⚙️) in the lower-left corner of the VSCode window and select **Settings**.
   - Alternatively, you can press `Ctrl+,` (Windows/Linux) or `Cmd+,` (Mac) to open the settings directly.

3. **Open `settings.json`**:
   - In the top right of the Settings tab, click on the `{}` icon to open the `settings.json` file. This provides a direct view of your settings file where you can make manual changes.

4. **Add the `files.associations` Setting**:
   - In the `settings.json` file, add the following configuration:
     ```json
       "files.associations": {
         "*.css": "tailwindcss"
       }
     ```
   - If there are already other settings in the file, just add this block within the existing JSON structure, ensuring proper syntax (commas between entries, etc.).

5. **Save the File**:
   - Save the changes by pressing `Ctrl+S` (Windows/Linux) or `Cmd+S` (Mac).

6. **Verify Changes**:
   - Open a `.css` file in VSCode to ensure that TailwindCSS syntax highlighting and features are now applied.

### Notes:
- The `files.associations` setting tells VSCode to treat files with a `.css` extension as TailwindCSS files, which can improve syntax highlighting and linting support for TailwindCSS classes.
- Make sure you have the TailwindCSS extension installed in VSCode for full functionality.

This approach helps tailor your editor to better support TailwindCSS-specific syntax and features in `.css` files.

**Reply by Frank67steven** [reply]
thanks  a lot this helped

**Reply by zaighamhussain786** [reply]
@Frank67steven You're welcome keep learning and working

**Reply by Sushant-Coder-01** [reply]
Thank you so much 😊

**Reply by zaighamhussain786** [reply]
> Thank you so much 😊

**Reply by SaipriyaPabba** [reply]
Thanks alot. It helped.

**Reply by zaighamhussain786** [reply]
> Thanks alot. It helped.

You're welcome.
Keep learning and growing ✨️

**Reply by andresmateo** [reply]
@tailwind base;
@tailwind components;
@tailwind utilities;
This code does not take effect in the browser despite all the changes. Please!!!! Help me. Thank you everyone The problem was solved.

**Reply by zaighamhussain786** [reply]
> @tailwind base; @tailwind components; @tailwind utilities; This code does not take effect in the browser despite all the changes. Please!!!! Help me.

If you’re sure that everything is ok so now you only need to restart your server or close everything and reopen that's very simple

**Reply by hw3727** [reply]
use Tailwind CSS IntelliSense

**Reply by ADTC** [reply]
Worth noting that if only some of your projects use Tailwind, it maybe better to add the setting in workspace level (that is, `.vscode/settings.json` at the project root) instead of user level.

**Reply by Ran-jeet01** [reply]
thank you sir

**Comment by zaighamhussain786** [comment]
Thank you 😊

**Comment by fuyalasmit** [comment]
You can fix this by setting up a new language mode.
Here's how you can do it in vs code:

 **1. Press ctrl + shift + p** 

 **2. Click on change language Mode**

 **3. Choose or type Tailwind CSS**

Also, make sure you are setting language mode when you're currently in index.css file.


Hope it helps, thank you!

**Reply by ADTC** [reply]
You need to install [Tailwind CSS Intellisense extension](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss) first.

**Reply by fireflysemantics** [reply]
THAT ROCKED - THANKS!!
